assignment1/src/main/java/ro/utcn/sd/he/assignment1/controller/CommandLineController.java

```diff
@@ -86,24 +89,160 @@ private boolean handleCommand(String command){
                 }
                 return false;
 
+            case "listAnswers":
+                handleListAnswers();
+                return false;
+
+            case "deleteAnswer":
+                handleDeleteAnswer();
+                return false;
+
+            case "edit":
+                handleEdit();
+                return false;
+
+            case "register":
+                handleRegister();
+                return false;
+
+            case "show":
+                handleShow();
+                return false;
+
+            case "answer":
+                handleAnswer();
+                return false;
             default:
                 System.out.println("Command unknwon");
                 return false;
         }
 
     }
 
+    private void handleListAnswers(){
+        answerService.findAllAnswers().forEach( a -> print(a.toString()));
+    }
+
+    private void handleEdit(){
+        if(isLogged()){
+            print("What answer do you want to edit (id):");
+            int answerId = scanner.nextInt();
+            scanner.nextLine(); //dump remaining line
+            Answer answer = answerService.findById(answerId);
+            if(userService.getAuthorOf(answer).getUsername().equals(user.getUsername())){
```

Comment:
The authorization checks are also business logic --> push them down to the service layer.

Ideally, you would have a method inside the answer service with the following signature: `Answer editAnswer(int answerId, int newText, User currentUser)`. All the authorization checks, loading the answer from the DB, updating and saving the answer would be located inside the service.

Reply:
That makes sense, I did as you adviced, thank you for pointing this out.